src/rust/bitbox02-rust/src/async_usb.rs

```diff
@@ -129,10 +189,20 @@ pub fn copy_response(dst: &mut [u8]) -> Result<usize, CopyResponseErr> {
     let mut state = USB_TASK_STATE.0.borrow_mut();
     match *state {
         UsbTaskState::Nothing => Err(CopyResponseErr::NotRunning),
-        UsbTaskState::Running(_) => Err(CopyResponseErr::NotReady),
+        UsbTaskState::Running(Some(_), ref mut next_request_state) => {
+            if let WaitingForNextRequestState::SendingResponse(ref response) = next_request_state {
+                let len = response.len();
+                dst[..len].copy_from_slice(&response);
+                *next_request_state = WaitingForNextRequestState::AwaitingRequest;
+                Ok(len)
+            } else {
+                Err(CopyResponseErr::NotReady)
```

Comment:
The docs mention `Err(false)` and `Err(true)` above. I think they're outdated.

Also, how about defining an `Error` enum with `CopyNotReady` and `CopyNotRunning`. I suspect we minght have other errors in the future, not just during "copy".

Reply:
Fixed the doc.

The second part: not sure, I usually don't like error enums where not every error is possible, cause then in the error handling you have a `_ => ...` arm that swallows new error conditions (where the compiler could warn you to handle them instead).

Anyway, I'll leave it as is for now, we don't even have more errors at the moment.